src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -132,7 +122,7 @@ class RelayDefaultNetworkLayer {
         method: 'POST',
       };
     }
-    return fetch(this._uri, init).then(throwOnServerError);
+    return fetch(this._uri, init).then(throwOnServerError.bind(this, request));
```

Comment:
Yeah, it wasn't a 100% ideal but that seemed to me the best way to share the error creation in all three places (or of course an arrow function if that has any advantages).